src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
@realshadow don't forget to update the lockfile to depend on the latest version of `composer/composer` when you submit a new PR (but please limit the update to `composer/composer` only by running `composer update composer/composer`). :)